src/tests/resistance/fuzzing.rs

```diff
@@ -0,0 +1,813 @@
+// Messages to be tested:
+// - Messages with any length and any content (random bytes).
+// - Messages with plausible lengths, e.g. 24 bytes for header and within the expected range for the body.
+// - Metadata-compliant messages, e.g. correct header, random body.
+// - Slightly corrupted but otherwise valid messages, e.g. N% of body replaced with random bytes.
+// - Messages with an incorrect checksum.
+// - Messages with differing announced and actual lengths.
+
+use crate::{
+    helpers::{autorespond_and_expect_disconnect, initiate_version_exchange},
+    protocol::{
+        message::*,
+        payload::{block::Headers, Addr, Nonce, Version},
+    },
+    setup::{config::read_config_file, node::Node},
+};
+
+use rand::{distributions::Standard, prelude::SliceRandom, thread_rng, Rng};
```

Comment:
It may be necessary to store the RNG seed for each trial - and/or the message contents required to reproduce the failure.

Since each run will be unique (I think?), if something does fail at some stage we will want to know what the seed for that instance was.

Reply:
Good point, I'll look into this 👍